### PR TITLE
[WFCORE-764] Remove the deprecated methods relating to ordered child …

### PR DESCRIPTION
**Pull request by kabir**
…resource. Hide the ordered child resource creator, this was simplified in PRs #813 and #821 

For this to work with the core integration tests, https://github.com/wildfly/wildfly/pull/7668 will need to be merged first.

### CODE REVIEW

**Comment by kabir** [comment]
Retest this please

**Comment by kabir** [comment]
Retest this please!

File: controller/src/main/java/org/jboss/as/controller/SimpleResourceDefinition.java [hunk 1]
```diff
@@ -441,16 +441,7 @@ public boolean isRuntime() {
      */
     @SuppressWarnings("deprecation")
```
Comment (bstansberry) [inline controller/src/main/java/org/jboss/as/controller/SimpleResourceDefinition.java]: Nit, only if you touch this for some reason: The SuppressWarnings looks to be cruft.

**Comment by bstansberry** [comment]
Other than the nit, which you can ignore, :+1:

**Pull request merged**